languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -424,6 +424,136 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+          <pattern>
+              <token inflected="yes">rentrer</token>
+              <token>à</token>
+              <token>l'</token>
+              <token>intérieur</token>
+          </pattern>
+          <message>Cette expression est un pléonasme.</message>
+          <suggestion>\1</suggestion>
+          <example correction="rentré">Je suis <marker>rentré à l'intérieur</marker>.</example>
+      </rule>
+      <rule>
+          <pattern>
+              <token>topographie</token>
+              <token>de</token>
+              <token>lieux</token>
+          </pattern>
+          <message>Cette expression est un pléonasme.</message>
+          <suggestion>\1</suggestion>
+          <example correction="topographie">La <marker>topographie de lieux</marker> est très utile pour concevoir des cartes.</example>
+      </rule>
+      <rule>
+          <pattern>
+              <token inflected="yes">suffire</token>
+              <token regexp="yes" min="0" max="1">pas|plus</token>
+              <token regexp="yes">juste|seulement</token>
+          </pattern>
+          <message>Cette expression est un pléonasme.</message>
+          <suggestion>\1</suggestion>
+          <example correction="suffit">Il <marker>suffit juste</marker> que tu prennes un manteau.</example>
+      </rule>
+      <rule>
+          <pattern>
+              <token regexp="yes">ne|n'</token>
+              <token postag="V.*" postag_regexp="yes"/>
+              <token>seulement</token>
+              <token>que</token>
+          </pattern>
+          <message>Cette expression est un pléonasme.</message>
+          <suggestion>\2 \3</suggestion>
+          <suggestion>\1\2 \4</suggestion>
+          <example correction="est seulement|n'est que">Il <marker>n'est seulement que</marker> midi.</example>
+      </rule>
+      <rule>
+          <antipattern>
+              <token>dans</token>
+              <token regexp="yes">laquelle|lequel|lesquel(le)?s</token>
+              <token>il</token>
+              <token>y</token>
+              <token inflected="yes">avoir</token>
+          </antipattern>
+          <pattern>
+              <token>dans</token>
+              <token regexp="yes">laquelle|lequel|lesquel(le)?s</token>
+              <token postag="R pers suj.*" postag_regexp="yes"/>
+              <token>y</token>
+          </pattern>
+          <message>Cette expression est un pléonasme.</message>
+          <suggestion>\1 \2 \3</suggestion>
+          <example correction="dans laquelle elle">Je n'ai pas regardé la série <marker>dans laquelle elle y</marker> joue.</example>
+      </rule>
+      <rule>
+          <pattern>
+              <token inflected="yes">pouvoir</token>
+              <token>peut-être</token>
+          </pattern>
+          <message>Cette expression peut être considérée comme un pléonasme.</message>
+          <suggestion>\1</suggestion>
+          <example correction="peux">Je <marker>peux peut-être</marker> l'aider.</example>
```

Comment:
So, my dissertation was about modality. That's why I am very keen on getting modal rules right. I don't agree with this one, it is not tautological.

peut-être: expresses epistemic possibility: the speaker considers something possible.
pouvoir faire qqch: expresses mostly ability, permission, and only sometimes epistemic possibility.

Bottom line: The two modals "peux" and "peut-être" express two different things. One is about the speaker's stance, the other one about the subject's abilities. It could be rephrased this way:

"Je suis possiblement capable de l'aider".

If you really want to keep this rule, you would have to make it about the phrasing. "Voulez-vous éviter les deux formes rapprochées de 'pouvoir' en substituant l'une d'entre elles ?" peux → sais; peut-être → possiblement

Reply:
I understand the difference therefore, this rule works only for some cases but they are more cases where it doesn't. The phrasing is good but it is likely it would not be helpful in most cases for the users.